### PR TITLE
Problem: nft module store is not initialized after upgrade

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- call SetStoreLoader to initialise nft module store

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by tomtau — APPROVED** [review]

File: app/app.go [hunk 1]
```diff
@@ -414,6 +415,27 @@ func New(
...
+
+	if upgradeInfo.Name == planName && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
+		storeUpgrades := storetypes.StoreUpgrades{
+			Added:   []string{nfttypes.StoreKey},
+			Renamed: []storetypes.StoreRename{},
+			Deleted: []string{},
+		}
+
+		// configure store loader that checks if version == upgradeHeight and applies store upgrades
+		app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
```
Comment (tomtau) [inline app/app.go:436]: should this just be called once / inside `SetUpgradeHandler` ?

**Review by devashishdxt — APPROVED** [review]

File: app/app.go [hunk 2]
```diff
@@ -414,6 +415,27 @@ func New(
 	)
 	app.SetEndBlocker(app.EndBlocker)
 
+	planName := "v2.0.0"
```
Comment (leejw51crypto) [inline app/app.go:418]: can this fetched from other place?
because two places hard-coded

File: app/app.go [hunk 3]
```diff
@@ -414,6 +415,27 @@ func New(
...
+	app.UpgradeKeeper.SetUpgradeHandler(planName, func(ctx sdk.Context, _ upgradetypes.Plan) {
+		// https://github.com/crypto-org-chain/chain-main/blob/master/doc/architecture/adr-003.md
+	})
+
+	upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
+	if err != nil {
+		panic(err)
+	}
+
+	if upgradeInfo.Name == planName && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
```
Comment (leejw51crypto) [inline app/app.go:428]: could add some explain for IsSkipHeight,
such as height > upgradeInfo.Height?

Comment (leejw51crypto) [reply app/app.go:428]: in latest planName is different from "2.0.0"? 
and not fall into this step?

File: app/app.go (hunk 1 above)
Comment (yihuang) [reply app/app.go:436]: I think the override only works before storage constructed, so it need to happen pretty early. while upgrade handler is called in abci event.

Comment (yihuang) [reply app/app.go:436]: I get the example code from the godoc of upgrade module: https://pkg.go.dev/github.com/cosmos/cosmos-sdk@v0.42.4/x/upgrade

File: app/app.go (hunk 2 above)
Comment (yihuang) [reply app/app.go:418]: I think the plan name only specified once here.

File: app/app.go (hunk 1 above)
Comment (yihuang) [reply app/app.go:436]: https://github.com/cosmos/cosmos-sdk/blob/master/x/upgrade/types/storeloader.go#L13
And before apply the upgrade, it'll check the height again, so it should be safe here.

File: app/app.go (hunk 3 above)
Comment (yihuang) [reply app/app.go:428]: cosmos-sdk has a feature that you can skip upgrades in command line:
```
$ chain-maind start --help
...
      --unsafe-skip-upgrades ints                       Skip a set of upgrade heights to continue the old binary
```

**Review by leejw51crypto — APPROVED** [review]
lgtm

**Pull request merged**